stripecardscan/src/main/java/com/stripe/android/stripecardscan/cardscan/CardScanActivity.kt

```diff
@@ -0,0 +1,683 @@
+import android.view.ViewGroup
+import android.view.WindowManager
+import androidx.annotation.Keep
+import androidx.annotation.StringRes
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+import androidx.core.view.updateMargins
+import com.stripe.android.camera.CameraAdapter
+import com.stripe.android.camera.CameraErrorListener
+import com.stripe.android.camera.CameraPreviewImage
+import com.stripe.android.camera.framework.AnalyzerLoopErrorListener
+import com.stripe.android.camera.framework.Stats
+import com.stripe.android.stripecardscan.R
+import com.stripe.android.stripecardscan.camera.getCameraAdapter
+import com.stripe.android.stripecardscan.cardscan.exception.InvalidStripePublishableKeyException
+import com.stripe.android.stripecardscan.cardscan.exception.UnknownScanException
+import com.stripe.android.stripecardscan.cardscan.result.MainLoopAggregator
+import com.stripe.android.stripecardscan.cardscan.result.MainLoopState
+import com.stripe.android.stripecardscan.cardscan.util.asRect
+import com.stripe.android.stripecardscan.cardscan.util.getColorByRes
+import com.stripe.android.stripecardscan.cardscan.util.getFloatResource
+import com.stripe.android.stripecardscan.cardscan.util.hide
+import com.stripe.android.stripecardscan.cardscan.util.setDrawable
+import com.stripe.android.stripecardscan.cardscan.util.setVisible
+import com.stripe.android.stripecardscan.cardscan.util.show
+import com.stripe.android.stripecardscan.cardscan.util.startAnimation
+import com.stripe.android.stripecardscan.databinding.ActivityCardscanBinding
+import com.stripe.android.stripecardscan.framework.Config
+import com.stripe.android.stripecardscan.framework.StorageFactory
+import com.stripe.android.stripecardscan.framework.util.getAppPackageName
+import com.stripe.android.stripecardscan.payment.card.ScannedCard
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlinx.parcelize.Parcelize
+import java.util.concurrent.atomic.AtomicBoolean
+import kotlin.coroutines.CoroutineContext
+import kotlin.math.min
+import kotlin.math.roundToInt
+
+internal const val INTENT_PARAM_REQUEST = "request"
+internal const val INTENT_PARAM_RESULT = "result"
+
+private val MINIMUM_RESOLUTION = Size(1067, 600) // minimum size of OCR
+private const val PERMISSION_RATIONALE_SHOWN = "permission_rationale_shown"
+
+sealed interface CancellationReason : Parcelable {
+
+    @Parcelize
+    object Closed : CancellationReason
+
+    @Parcelize
+    object Back : CancellationReason
+
+    @Parcelize
+    object CameraPermissionDenied : CancellationReason
+}
+
+internal interface ScanResultListener {
+
+    /**
+     * The scan completed.
+     */
+    fun cardScanComplete(pan: String)
```

Comment:
We may want to add more fields to the card in the future, like name, expiry, etc. Can this return a `ScannedCard` object rather than just the pan?